controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +247,62 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    /*
+        Conditional delete of the specified key with the specified version.
+        If the conditional delete fails it is ignored.
+     */
+    private CompletableFuture<Void> conditionalDeleteOfKey(String tableName, long requestId, TableSegmentKey key,
```

Comment:
we have a method for condition delete of key called - removeEntry where if you provide a version then it is used to delete the key.
is there a need for this new method?

Reply:
Yes, The main differences with removeEntry method are 
- if the conditional delete fails due to a concurrent update this error is ignored. 
- cache is not used for the tag values while the removeEntry method uses that.

Reply:
remove entry simply removes from the cache as well if it exists. so its safe.

remove entry will throw the concurrency update error. if supplied key version is null, only in that case it does unconditional delete. otherwise it does conditional delete. 

https://github.com/pravega/pravega/blob/9b4e6156390d2e2c9e9897efd4687773e9c9ed6e/controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java#L562

it does handle the idempotent delete when it gets dataNotFound exception and converts it to a happy case.

Reply:
Thanks, I have improved the code to remove this code duplication.